src/main/scala/io/ino/solrs/SolrServers.scala

```diff
@@ -54,7 +54,15 @@ trait SolrServers {
 }
 
 class StaticSolrServers(override val all: IndexedSeq[SolrServer]) extends SolrServers {
-  override def matching(r: SolrRequest[_]): Try[IndexedSeq[SolrServer]] = Success(all)
+  override def matching(r: SolrRequest[_]): Try[IndexedSeq[SolrServer]] = {
+    if(!Utils.isAdminType(r)) {
+      Success(all)
+    } else {
+      Success(all.map(solrServer => {
+        SolrServer(Utils.solrUrl(solrServer.baseUrl), solrServer.status)
+      }))
```

Comment:
Same here: I don't understand why this is necessary. And I don't think that it should be the responsibility of `SolrServers` to change the url of the returned solr server depending on the context.